File: C++/434.cpp
__________________________________________________________________________________________________
sample 4 ms submission
class Solution {
public:
    int countSegments(string s) {
        int count=0;
        if(s=="")
            return 0;
        s.push_back(' ');
        for(int i=1;i<s.size();i++)
        {
            if(s.at(i)==' ')
            {
                count++;
                if(s.at(i-1)==' ')
                    count--;
            }
        }
        return count;
    }
};
__________________________________________________________________________________________________
× Close
sample 8328 kb submission
class Solution {
public:
    int countSegments(string s) {
        int count = 0;
        char pre = ' ';
        for (char cur : s) {
            if (pre == ' ' && cur != ' ') count++;
            pre = cur;
        }
        return count;
    }
};
__________________________________________________________________________________________________

File: Java/434.java
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
    public int countSegments(String s) {
        boolean space = true;
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') space = true;
            else if (space) {
                count++;
                space = false;
            }
        }
        return count;
    }
}
__________________________________________________________________________________________________
sample 33232 kb submission
class Solution {
    public int countSegments(String s) {
        return (int) Arrays.stream(s.split(" ")).filter(item -> !item.isEmpty()).count();
    }
}
__________________________________________________________________________________________________
